tests/pyanaconda_tests/kickstart_dispatcher.py

```diff
@@ -0,0 +1,486 @@
+
+        elements = TrackedKickstartElements()
+        elements.append_section(element1.args[0], element1.content, element1.lineno, filename)
+        elements.append_command(element2.args[0], element2.content, element2.lineno, filename)
+        elements.append_command(element3.args[0], element3.content, element3.lineno, filename)
+        elements.append_addon(element4.args[1], element4.content, element4.lineno, filename)
+        elements.append_command(element5.args[0], element5.content, element5.lineno, filename)
+        elements.append_addon(element6.args[1], element6.content, element6.lineno, filename)
+        elements.append_section(element7.args[0], element7.content, element7.lineno, filename)
+
+        ### Filtering elements
+
+        all_elements = elements.all_elements
+        self.assertEqual(all_elements, expected_elements)
+
+        # elements of specified type and filter are returned
+        network_commands = elements.get_elements(commands=["network"])
+        self.assertEqual(network_commands, [element2, element3])
+        pony_addon = elements.get_elements(addons=["pony"])
+        self.assertEqual(pony_addon, [element4])
+        self.assertEqual(elements.get_elements(), [])
+
+        # to get all elements of specified type use empty list as filter
+        all_addons_and_sections = elements.get_elements(sections=[], addons=[])
+        self.assertEqual(all_addons_and_sections, [element1, element4, element6, element7])
+
+        ### Tracking elements
+
+        processed_elements = elements.get_elements(commands=["network"], sections=["%pre"], addons=["pony"], process=True)
+        unprocessed_elements = elements.unprocessed_elements
+        # still keeping order of elements
+        self.assertEqual(unprocessed_elements, [element5, element6, element7])
+        # nothing is missing
+        self.assertEqual(set(all_elements), set.union(set(processed_elements), set(unprocessed_elements)))
+        # elements once processed remain processed if you get them without processing
+        elements.get_elements(commands=["network"], sections=["%pre"], addons=["pony"])
+        self.assertEqual(elements.unprocessed_elements, unprocessed_elements)
+        # process=False is just default (NOOP)
+        elements.get_elements(commands=["network"], sections=["%pre"], addons=["pony"], process=False)
+        self.assertEqual(elements.unprocessed_elements, unprocessed_elements)
+        # processing some more elements - firewall command
+        firewall_elements = elements.get_elements(commands=["firewall"], process=True)
+        self.assertEqual(set(elements.unprocessed_elements), set.difference(set(unprocessed_elements), set(firewall_elements)))
+        self.assertEqual(elements.unprocessed_elements, [element6, element7])
+        # now process everthing
+        elements.get_elements(commands=[], sections=[], addons=[], process=True)
+        self.assertEqual(elements.unprocessed_elements, [])
```

Comment:
Please split these to different test methods. These are already different tests but instead of `def...` they have comments between them.